msteams-platform/tabs/how-to/using-teams-client-sdk.md

```diff
@@ -1,72 +1,469 @@
+* **Don't** use *hostName* to gate API calls. Instead, check for capability support (`isSupported`).
+* **Do** use *hostName* to differentiate the theme of your application based on the host it's running in. For example, you can use Microsoft Teams purple as the main accent color when running in Teams, and Outlook blue when running in Outlook.
+* **Do** use *hostName* to differentiate messages shown to the user based on which host it's running in. For example, show *Manage your tasks in Office* when running in Office on the web, and *Manage your tasks in Teams* when running in Microsoft Teams.
+
+#### Namespaces
+
+Starting with TeamsJS v.2.0, APIs are organized into *capabilities* by way of namespaces. Several new namespaces of particular importance are *app*, *pages*, *dialog*, and *teamsCore*.
+
+##### *app* namespace
+
+The `app` namespace contains top-level APIs required for overall app usage, across Microsoft Teams, Office, and Outlook. All the APIs from various other TeamsJS namespaces have been moved to the `app` namespace as of TeamsJS v.2.0:
+
+| Original namespace `global (window)` | New namespace `app` |
+| - | - |
+| `initialize` | `app.initialize` |
+| `getContext` | `app.getContext` |
+| `registerOnThemeChangeHandler` | `app.registerOnThemeChangeHandler` |
+
+| Original namespace `appInitialization` | New namespace `app` |
+| - | - |
+| `appInitialization.notifyAppLoaded` | `app.notifyAppLoaded` |
+| `appInitialization.notifySuccess` | `app.notifySuccess` |
+| `appInitialization.notifyFailure` | `app.notifyFailure` |
+| `appInitialization.notifyExpectedFailure` | `app.notifyExpectedFailure` |
+| `appInitialization.FailedReason` enum | `app.FailedReason` |
+| `appInitialization.ExpectedFailureReason` enum | `app.ExpectedFailureReason` |
+| `appInitialization.IFailedRequest` enum | `app.IFailedRequest` |
+| `appInitialization.IExpectedFailureRequest` enum | `app.IExpectedFailureRequest` |
+
+##### *pages* namespace
+
+The `pages` namespace includes functionality for running and navigating webpages within various Microsoft 365 hosts, including Teams, Office, and Outlook. It also includes several subcapabilities, implemented as subnamespaces.
+
+| Original namespace `global (window)` | New namespace `pages` |
+| - | - |
+| `setFrameContext` | `pages.setCurrentFrame` (renamed) |
+| `initializeWithFrameContext` | `pages.initializeWithFrameContext` |
+| `registerFocusEnterHandler` | `pages.registerFocusEnterHandler`
+| `registerFullScreenHandler` | `pages.registerFullScreenHandler` |
+| `navigateCrossDomain` | `pages.navigateCrossDomain` |
+| `returnFocus` | `pages.returnFocus` |
+
+###### *pages.tabs*
+
+| Original namespace `global (window)` | New namespace `pages.tabs` |
+| - | - |
+| `getTabInstances` |  `pages.tabs.getTabInstances` |
+| `getMruTabInstances` | `pages.tabs.getMruTabInstances` |
+| `navigateToTab` | `pages.tabs.navigateToTab` |
```

Comment:
This looks like it's here twice? It was only originally in TeamsJS v1 as `navigation.navigateToTab`, not in the global namespace